src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -96,22 +105,36 @@ interface IProps {
   setDateCallback?: (date: Date) => void
   setTimeCallback?: (date: Date) => void
   allowAfterMidnight?: boolean,
+  checkTimeValidation?: boolean,
 }
 
 const DateTimeBlock : React.FC<IProps> = ({
   allowAfterMidnight = false,
   title,
   hasDate,
   hasTime,
+  checkTimeValidation,
   date = new Date(),
   setDateCallback = ()=>{},
 }) => {
 
-
-  const [displayHours, setDisplayHours] = useState<string>(format(date, "mm"));
-  const [displayMinutes, setDisplayMinutes] = useState<string>(format(date,'HH'));
+  const convertHours = (date.getHours()).toString();
+  const convertMinutes = date.getMinutes().toString();
+  const [displayHours, setDisplayHours] = useState<string>(convertHours);
+  const [displayMinutes, setDisplayMinutes] = useState<string>(convertMinutes);
 
   const setDateHours = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
+    const isContainsSymbolN =  /^[0-9]+[0-9]*$/;
+    if(value === '000' || value.length>2){
+      return;
+    } else if (Number(value)>24){
+      value = '24';
+    } else if (!isContainsSymbolN.test(value)) {
+      if(value !== ''){
+        return ;
+      }
+    }
+    setDisplayHours(value);
```

Comment:
You can use RegEx here as well
**RegEx**:  `(^[0-1]{0,1}[0-9]$)|(^2[0-4]$)`

Reply:
Done